### PR TITLE
add model for logging indexes

### PR DESCRIPTION
**Pull request by smo921**
Datadog has recently released the public beta of the API for managing logging indexes.  It would be great to support this in Kennel.

https://docs.datadoghq.com/api/?lang=python#logs-indexes

= TODO
- [ ] Update generator / sync process to include index configuration and index order

### CODE REVIEW

File: lib/kennel/models/logs_index.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# Manage logging index configuration and ordering
+# https://docs.datadoghq.com/api/?lang=python#logs-indexes
+module Kennel
+  module Models
+    class LogsIndex < Base
+      @@index_order = []
```
Comment (grosser) [inline lib/kennel/models/logs_index.rb]: not used ?

**Comment by grosser** [comment]
sounds reasonable ... doubt it works in the current state
not a big fan of the `index_order` since it will make testing hard and in general will be brittle/unexpected :(

**Comment by smo921** [comment]
I can ditch the index ordering and use an array, instead of hiding the fact that ordering matters.

**Comment by grosser** [comment]
why do they need to be ordered ?
... use the api like all the other models ?

**Comment by smo921** [comment]
The order of the indexes in the configuration matters because log events are placed into indexes on a first-match based on the index filter.

https://docs.datadoghq.com/api/?lang=bash#update-indexes-order

**Comment by grosser** [comment]
let the users set something like `z-index` so they can adjust the level they want to be in ?
otherwise sort alphabetically ...

**Comment by smo921** [comment]
This is probably all pointless anyway.  If everyone can define a single attribute in their logs to route them to the correct index them the configuration is simple and none of this is necessary.

**Comment by grosser** [comment]
yes, let's start with that :D

**Pull request closed without merging**